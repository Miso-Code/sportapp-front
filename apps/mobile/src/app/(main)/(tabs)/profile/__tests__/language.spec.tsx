import React from 'react'
import renderer, { ReactTestRenderer, act } from 'react-test-renderer'

import Language from '../language'
import { useTranslation } from 'react-i18next'

jest.mock('expo-router')

jest.mock('@sportapp/stores', () => ({
	useUserStore: jest.fn()
}))

describe('Language', () => {
	let component: ReactTestRenderer

	beforeEach(() => {
		jest.clearAllMocks()
		component = renderer.create(<Language />)
	})

	afterEach(() => {
		component.unmount()
	})

	it('should render correctly', () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should render 2 language options', () => {
		const languages = component.root.findAllByProps({ testID: 'language' })
		expect(languages).toHaveLength(4) // react native makes it two per instance
	})

	it('should render active language', () => {
		const languages = component.root.findAllByProps({ testID: 'language' })
		expect(languages[0].props.children[0].props.status).toBe('unchecked')
		expect(languages[2].props.children[0].props.status).toBe('checked')

		const { i18n } = useTranslation()
		i18n.changeLanguage('es')

		act(() => {
			component.update(<Language />)
		})

		const languagesUpdated = component.root.findAllByProps({
			testID: 'language'
		})
		expect(languagesUpdated[0].props.children[0].props.status).toBe(
			'checked'
		)
		expect(languagesUpdated[2].props.children[0].props.status).toBe(
			'unchecked'
		)
		i18n.changeLanguage('en')
	})

	it('should update language when a language text is pressed', async () => {
		const languages = component.root.findAllByProps({ testID: 'language' })
		await act(async () => {
			languages[0].props.children[1].props.onPress()
		})
		const { i18n } = useTranslation()
		expect(i18n.language).toBe('es')

		act(() => {
			languages[2].props.children[1].props.onPress()
		})
		const { i18n: i18nUpdated } = useTranslation()

		expect(i18nUpdated.language).toBe('en')
	})

	it('should update language when a radio button group changes', async () => {
		const radioGroup = component.root.findByProps({
			'data-testID': 'radio-group'
		})
		await act(async () => {
			radioGroup.props.onValueChange('es')
		})
		const { i18n } = useTranslation()
		expect(i18n.language).toBe('es')

		act(() => {
			radioGroup.props.onValueChange('en')
		})
		const { i18n: i18nUpdated } = useTranslation()

		expect(i18nUpdated.language).toBe('en')
	})
})
