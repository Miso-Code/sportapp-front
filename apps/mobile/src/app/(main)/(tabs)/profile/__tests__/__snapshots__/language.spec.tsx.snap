// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "justifyContent": "flex-start",
      "marginBottom": 15,
      "marginHorizontal": 15,
      "marginTop": 120,
    }
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "writingDirection": "ltr",
        },
        [
          {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0.25,
            "lineHeight": 20,
          },
          undefined,
        ],
      ]
    }
  >
    language.changeText
  </Text>
  <View
    accessibilityRole="radiogroup"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 10,
        }
      }
      testID="language"
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": false,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "borderColor": "rgba(73, 69, 79, 1)",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        onPress={[Function]}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            undefined,
          ]
        }
      >
        language.options.es
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 10,
        }
      }
      testID="language"
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="radio"
        accessibilityState={
          {
            "busy": undefined,
            "checked": true,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 18,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "borderColor": "rgba(103, 80, 164, 1)",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "margin": 8,
              "width": 20,
            }
          }
        >
          <View
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "rgba(103, 80, 164, 1)",
                  "borderRadius": 5,
                  "height": 10,
                  "transform": [
                    {
                      "scale": 1,
                    },
                  ],
                  "width": 10,
                }
              }
            />
          </View>
        </View>
      </View>
      <Text
        onPress={[Function]}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            undefined,
          ]
        }
      >
        language.options.en
      </Text>
    </View>
  </View>
</View>
`;
